script.js

```diff
@@ -0,0 +1,578 @@
+  },
+
+
+  // method delete symbol in textAria
+  typeDelete() {
+    const text = this.elements.textAria.value;
+    const st = this.elements.textAria.selectionStart;
+    const end = this.elements.textAria.selectionEnd;
+
+    if (st === end) {
+      this.elements.textAria.value = text.substring(0, st) + text.substring(end + 1);
+      this.elements.textAria.selectionStart = st;
+      this.elements.textAria.selectionEnd = st;
+    } else {
+      this.elements.textAria.value = text.substring(0, st) + text.substring(end);
+      this.elements.textAria.selectionStart = st;
+      this.elements.textAria.selectionEnd = st;
+    }
+    this.elements.textAria.focus();
+  },
+
+
+  // method toggle CapsLock
+  toggleCapsLock() {
+    const elm = this.elements.keyBox.querySelector('button[data-key-code="CapsLock"]');
+    if (!this.inputAria.capsLockToggle) {
+      elm.classList.add('turn-on');
+      this.setCapsLock();
+      this.inputAria.capsLockToggle = true;
+    } else {
+      elm.classList.remove('turn-on');
+      this.setShiftUpLayout();
+      this.inputAria.capsLockToggle = false;
+    }
+  },
+
+
+  // method handle FunctionKeys
+  handleFunctionKeys(keyCode) {
+    switch (keyCode) {
+      case 'BackSpace':
+        this.typeBackspace();
+        break;
+      case 'Delete':
+        this.typeDelete();
+        break;
+      case 'Enter':
+        this.type('\n');
+        break;
+      case 'Tab':
+        this.type('\t');
+        break;
+      case 'ShiftLeft':
+        this.setShiftDownLayout();
+        break;
+      case 'ShiftRight':
+        this.setShiftDownLayout();
+        break;
+      case 'CapsLock':
+        this.toggleCapsLock();
+        break;
+      // no default
+    }
+  },
+
+
+  // method handle events of keyBoard: keydown, 'keyup'
+  handleEvent(event) {
+    const elm = this.elements.keyBox.querySelector(`button[data-key-code=${event.code}]`);
+    if (!elm) return;
+
+    switch (event.type) {
+      case 'keydown':
+        elm.classList.add('active');
+        if (!(this.elements.functionKeys.includes(elm.dataset.keyCode))) {
+          const char = elm.textContent;
+          this.type(char);
+        }
+        if (this.elements.functionKeys.includes(elm.dataset.keyCode)) {
+          this.handleFunctionKeys(elm.dataset.keyCode);
+        }
+        if ((event.ctrlKey && event.altKey)) { // Ctrl+Shift - switch language
+          this.switchLanguageLayout();
+        }
+        event.preventDefault();
+        break;
+
+      case 'keyup':
+        elm.classList.remove('active');
+        if (elm.dataset.keyCode === 'ShiftLeft' || elm.dataset.keyCode === 'ShiftRight') {
+          this.setShiftUpLayout();
+        }
+        break;
+      // no default
+    }
+  },
+
+
+  // initialization object KeyBoard
+  init() {
+    // localStorage
+    if (!(Object.prototype.hasOwnProperty.call(localStorage, 'virtualKeyboardLang'))) {
+      // if (localStorage.hasOwnProperty('virtualKeyboardLang')) {
```

Comment:
Should not be commented code in final PR

Reply:
Done

Reply:
:thumbsup: